### PR TITLE
add preheat implements

### PR DESCRIPTION
**Pull request by ansinlee**

### CODE REVIEW

**Comment by Starnop** [comment]
@jim3ma PTAL. Thanks.

File: supernode/server/preheat_bridge.go [hunk 1]
```diff
@@ -60,8 +61,8 @@ func (s *Server) getPreheatTask(ctx context.Context, rw http.ResponseWriter, req
 	}
 	resp := types.PreheatInfo{
 		ID:         task.ID,
-		FinishTime: strfmt.NewDateTime(),
-		StartTime:  strfmt.NewDateTime(),
+		FinishTime: strfmt.DateTime(time.Unix(task.FinishTime/1000, task.FinishTime%1000*int64(time.Millisecond)).UTC()),
```
Comment (Starnop) [inline supernode/server/preheat_bridge.go:64]: Change type of `FinishTime` form int64 to time.Time? WDYT?

File: supernode/daemon/mgr/preheat/preheat_task_repository.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+import (
+	"sync"
+	"time"
+
+	"github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr"
+)
+
+const(
+	EXPIRED_TIME = 7 * 24 * 3600 * 1000;
```
Comment (Starnop) [inline supernode/daemon/mgr/preheat/preheat_task_repository.go:26]: Please add a comment here.

File: supernode/daemon/mgr/preheat/preheat_task_repository.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	}
+	return t != nil && r.expired(t.(*mgr.PreheatTask).StartTime)
+}
+
+func (r *PreheatTaskRepository) expired(timestamp int64) bool {
+	return time.Now().UnixNano()/int64(time.Millisecond) > timestamp+EXPIRED_TIME
+}
+
+func(r *PreheatTaskRepository) load() {
+	r.loaded.Do(func(){
```
Comment (Starnop) [inline supernode/daemon/mgr/preheat/preheat_task_repository.go:117]: useless code？

File: supernode/daemon/mgr/preheat/file_preaheater.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	"github.com/sirupsen/logrus"
+	"time"
+
+	"github.com/dragonflyoss/Dragonfly/apis/types"
+	"github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr"
+)
+
+func init() {
+	RegisterPreheater("file", &FilePreheat{BasePreheater:new(BasePreheater)})
+	logrus.StandardLogger().SetLevel(logrus.DebugLevel)
```
Comment (Starnop) [inline supernode/daemon/mgr/preheat/file_preaheater.go:29]: test code?

File: supernode/daemon/mgr/preheat/preheat_service.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+	if r, ok := header["Range"]; ok {
+		id = fmt.Sprintf("%s%s%s%s%s", key, url, sign, r, key)
+	} else {
+		id = fmt.Sprintf("%s%s%s%s", key, url, sign, key)
+	}
+	return digest.Sha256(id)
+}
+
+func (svc *PreheatService) createCommand(url string, header map[string]string, filter, identifier, tmpTarget string) *exec.Cmd {
+	netRate := 50
```
Comment (Starnop) [inline supernode/daemon/mgr/preheat/preheat_service.go:127]: please don't use magic numbers? It is easy to make others confused.

File: supernode/daemon/mgr/preheat/preheat_task_repository.go (hunk 3 above)
Comment (ansinlee) [reply supernode/daemon/mgr/preheat/preheat_task_repository.go]: removed

File: supernode/daemon/mgr/preheat/preheat_task_repository.go (hunk 2 above)
Comment (ansinlee) [reply supernode/daemon/mgr/preheat/preheat_task_repository.go:27]: done

**Comment by garfield009** [comment]
agree

**Pull request merged**